internal/executor/utilisation/cluster_utilisation.go

```diff
@@ -165,24 +172,52 @@ func (clusterUtilisationService *ClusterUtilisationService) GetAllNodeGroupAlloc
 		return []*NodeGroupAllocationInfo{}, err
 	}
 
+	batchPods, err := clusterUtilisationService.clusterContext.GetBatchPods()
+	if err != nil {
+		// Failing this step will just mean cordoned utilisation won't be accounted for
+		// This isn't a critical failure so we log and continue
+		log.Errorf("Error fetching batch pods to calculate cordond node utilisation: %v", err)
+		batchPods = []*v1.Pod{}
```

Comment:
Probably simpler to just return error in the case that call fails. If it fails it is probably indicative of a wider problem.

If you have a fair chunk of the cluster cordoned, it could be quite wrong to not report cordoned. Simpler to just have it all succeed or all fail, rather than have partial failures.